### PR TITLE
Add core concepts search jto

### PR DESCRIPTION
**Pull request by jobannon**
# Summary
- Add Search to core concepts
- Most of the content was divorced/broken-out from the `Users Core Concept`
- Keeps our documentation `D.R.Y` (avoid duplication of documentation under `Entities` as they will have search functionality as well)

### CODE REVIEW

File: site/docs/v1/tech/core-concepts/users.adoc [hunk 1]
```diff
@@ -42,28 +42,28 @@ include::docs/v1/tech/shared/_login_events.adoc[]
...
 == User Search
 
-User search requests may be made through the link:/docs/v1/tech/apis/users#search-for-users[User Search API] or within the FusionAuth admin UI under [breadcrumb]#Users#.
+[NOTE.since]
+====
 
 As of version 1.16.0, FusionAuth ships with a database search engine as the default.
 
-=== Configuration
+By selecting the appropriate link:/docs/v1/tech/installation-guide/fast-path[Fast Path installation], one can easily download a configuration with Elasticsearch pre-enabled.
```
Comment (mooreds) [inline site/docs/v1/tech/core-concepts/users.adoc:50]: I would leave off fastpath, but point to the install guide section if you want.

File: site/docs/v1/tech/core-concepts/users.adoc [hunk 2]
```diff
@@ -42,28 +42,28 @@ include::docs/v1/tech/shared/_login_events.adoc[]
...
 
-See the [field]#fusionauth-app.search-engine-type# and [field]#fusionauth-app.search-servers# properties, as well as the `FUSIONAUTH_SEARCH_ENGINE_TYPE` and `FUSIONAUTH_SEARCH_SERVERS` environment variables definitions of the link:/docs/v1/tech/reference/configuration/[Configuration Reference] for details on configuring the search engine and (optional) Elasticsearch integration.
+You can read more in our link:/docs/v1/tech/core-concepts/search[search core concept]
```
Comment (mooreds) [inline site/docs/v1/tech/core-concepts/users.adoc:52]: Would change to -> "You can read more about the database and other search engine in the link:...[search core concepts section]."

File: site/docs/v1/tech/core-concepts/users.adoc [hunk 3]
```diff
@@ -74,21 +74,9 @@ The database search engine enables fuzzy search against the following fields of
 
 image::user-search-database.png[User Search with Database Search Engine,width=1200,role=shadowed bottom-cropped]
```
Comment (mooreds) [inline site/docs/v1/tech/core-concepts/users.adoc:75]: Would put another link:

To learn more about the database search engine in general, view the link:...[search core concepts section].

File: site/docs/v1/tech/core-concepts/users.adoc [hunk 4]
```diff
@@ -74,21 +74,9 @@ The database search engine enables fuzzy search against the following fields of
...
-All search terms are converted to lowercase and compared with lowercase values.
-
-Regular expressions, ranges and other complicated queries can not be used.
-
 === Elasticsearch Search Engine
 
-Leveraging Elasticsearch for the user search engine, enables advanced search capabilities on more numerous and granular data and a performance improvement for user search.
-
-The Elasticsearch search engine is appropriate for systems that are dependent on the link:/docs/v1/tech/apis/users#search-for-users[User Search APIs], are expected to have a large number of active users, and requires a more tactical search than is provided by the database search engine.
+Leveraging Elasticsearch for the user search engine enables advanced search capabilities on more numerous and granular data and a performance improvement for user search.
```
Comment (mooreds) [inline site/docs/v1/tech/core-concepts/users.adoc:79]: " and granular data and a performance improvement" -> " and granular data. It also provides a performance improvement."

File: site/docs/v1/tech/core-concepts/users.adoc [hunk 5]
```diff
@@ -107,12 +95,3 @@ When searching for users by application or any fields on an application, it is n
 The following screenshot shows an Elasticsearch JSON query being constructed to search for users that match the email pattern `*@fusionauth.io`, are registered to the `Pied Piper` application, and are assigned the `admin` role:
 
 image::user-search-json-query.png[User Search by JSON Query,width=1200,role=shadowed top-cropped]
```
Comment (mooreds) [inline site/docs/v1/tech/core-concepts/users.adoc:97]: Would put another link:

To learn more about the Elasticsearch search engine in general, view the link:...[search core concepts section].

File: site/docs/v1/tech/core-concepts/search.adoc [hunk 6]
```diff
@@ -0,0 +1,92 @@
...
+
+By selecting the appropriate link:/docs/v1/tech/installation-guide/fast-path[Fast Path installation], one can easily download a configuration with Elasticsearch pre-enabled.
+
+====
+
+==== Configuration Reference
+
+See link:/docs/v1/tech/reference/configuration/[Configuration Reference] for details on configuring the search engine and (optional) Elasticsearch integration.
+
+Highlighted here:
```
[the remark this answers is not in the document]
Comment (jobannon) [reply site/docs/v1/tech/core-concepts/search.adoc:31]: Good point!

File: site/docs/v1/tech/core-concepts/search.adoc [hunk 7]
```diff
@@ -0,0 +1,92 @@
...
+
+==== Switch engine types
+
+You may also link:/docs/v1/tech/tutorials/switch-search-engines/[switch between the different search engines].
+
+==== Docker
+If you are running FusionAuth in a Docker environment, see the link:/docs/v1/tech/installation-guide/docker/[Using FusionAuth on Docker] documentation for an example configuring Elasticsearch as the user search engine.
+
+
+==== View current configuration
```
[the remark this answers is not in the document]
Comment (jobannon) [reply site/docs/v1/tech/core-concepts/search.adoc:51]: @mooreds 

Sweet!  Checked all others.  Should I add a section for `docs` then in the DocsDevReadme?  

I was going off of blog posts RE: assumptions
`- Blog post headers should have only the first word and any proper nouns are capitalized.`

File: site/docs/v1/tech/core-concepts/search.adoc [hunk 8]
```diff
@@ -0,0 +1,92 @@
...
+
+Regular expressions, ranges, and other complicated queries can not be used.
+
+== Using the Elasticsearch search engine
+
+Leveraging Elasticsearch, enables advanced search capabilities on more numerous and granular data and a performance improvement.
+
+The Elasticsearch search engine is appropriate for systems that are dependent on the FusionAuth API's (such as link:/docs/v1/tech/apis/users#search-for-users[users]), are expected to have a large number of results, and requires a more tactical search than is provided by the database search engine.
+
+=== Reindex
```
[the remark this answers is not in the document]
Comment (jobannon) [reply site/docs/v1/tech/core-concepts/search.adoc:79]: I don't think anything else had just `reindex` in the title.  

Closest was:
`Verify the change and reindex` in a tutorial.

**Pull request merged**